Solution1/Exercise1/Program.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+
+// Последовательно считать с консоли 3 числа (x y z)
+// Вычислить и вывести на экран результаты следующих выражений:
+// 1. Сумма x и y, умноженная на увеличенный на единицу квадрат z
+// 2. Уменьшенный на единицу остаток от деления x на z, умноженный на
+// квадратный корень из y
+// 3. Сумма умножений пар чисел х и y & y и z, разделенная на куб числа z
+
+namespace Exercise1
+{
+    class Program
+    {
+        static double? ConvertInsertedValueToInteger(string insertedValue)
+        {    
+            return double.TryParse(insertedValue, out var temp) ? double.Parse(insertedValue) : (double?)null; 
```

Comment:
Your function is working, but too hard to understand and support by another developer
you have nullable function, with do-while, with ternary operator and its just for cycle for parsing symbol to the double.
You could easily remove your cycles and if check, if you function will be easier.
what you should do:
1. Write user that he need to input symbol 1 time
2. try to parse his value, if success - go further, if no - say it to him and ask again
Try to use return value of the TryParse function and out parameter of it, using first you could do while cycle, and second will be your parsed value.
If there would be problems we could discuss it

Reply:
Done